multiqc/modules/adapterRemoval/adapterRemoval.py

```diff
@@ -0,0 +1,297 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Adapter Removal """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+
+from multiqc import config
+from multiqc.plots import bargraph, linegraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Adapter Removal',
+        anchor='adapterRemoval', target='Adapter Removal',
+        href='https://github.com/MikkelSchubert/adapterremoval',
+        info=" rapid adapter trimming, identification, and read merging ")
+
+        self.__read_type = None
+        self.__collapsed = None
+        self.s_name = None
+        self.adapter_removal_data = {}
+
+        self.len_dist_plot_data = {
+            'mate1': dict(),
+            'mate2': dict(),
+            'singleton': dict(),
+            'collapsed': dict(),
+            'collapsed_truncated': dict(),
+            'discarged': dict(),
+            'all': dict(),
+        }
+
+        parsed_data = None
+        for f in self.find_log_files(config.sp['adapterRemoval'], filehandles=True):
```

Comment:
After refactoring, this now takes the search pattern string instead of the dict:

```python
for f in self.find_log_files('adapterRemoval', filehandles=True):
```